### ISSUE TITLE
BUG: Memory Leak

### ISSUE THREAD

**Issue by SonicRay100** [Ross-Patterson/Portals-Desktop-Organization#353]
### Check the Help Site and GitHub Issues/Discussions

- [X] I have checked and found nothing.

### Version

3.3.0.9

### Operating System

Windows 11

### What happened?

There is a small memory leak when using Portals.

- Moving cursor over items in a portal.
- Clicking on items in a portal.
- Creating a portal.
- Modifying a portal.
- Moving a portal.
- Applying a layout.

The leak is relatively small but can increase substantially when first setting up a portal, populating it, and configuring its appearance. Normally, Portals start out with about 130-140 MB of memory usage, but after a few modifications, I can see it using over 200 MB.

While this may not seem like a lot, it becomes far more noticeable when applying layouts. If you apply a layout a few times, the memory usage will increase by over 50 MB with each application, assuming the user has as many portals as I do (10). I stopped testing after seeing memory usage exceed 600 MB, as the leak doesn't seem to stop. Fortunately, this shouldn't be a significant issue, as most users won't be applying layouts continuously, and the memory leak can be resolved by simply restarting Portals.

This would be more problematic for users who put their computers to sleep or hibernate, rather than shutting down and flushing their memory. However, it’s easy to resolve by restarting Portals.

### Steps to Reproduce

1. Create several portals (the more, the better) and populate them with numerous items.
2. Move them around and modify them while monitoring Portals' memory usage in Task Manager or another application that can display an app's memory usage.
3. Save a layout and apply it to observe a larger memory leak.
4. Repeat step 3 to confirm that the memory leak increases with each application of a layout.

### What would you expect to happen?

Fix memory leak.

**Comment by thx1200** [comment]
I've noticed after it runs for days the memory usage keeps going up, this must be why.  I restart portals every few days.

**Comment by Ross-Patterson** [comment]
I will definitely check on this.

**Comment by gen2-sys-it** [comment]
Same here : 
![image](https://github.com/user-attachments/assets/719625ff-31c6-4572-a52b-38602e8617c5)

Memory use immediately jumps to >700MB on start???

**Comment by gen2-sys-it** [comment]
Almost 800MB being used right now. I have 3 folders/portals.. that's it, each with 6 or less items. The memory usage of this program is almost as bad as chrome! :)

Edit : Now at 836MB.

**Comment by gen2-sys-it** [comment]
@Ross-Patterson any chance you can look into this? Still hovering about 800MB usage all the time.
![image](https://github.com/user-attachments/assets/62171105-43f8-451c-a2b5-6a990150d44b)

**Comment by Ross-Patterson** [comment]
@gen2-sys-it any specific action being performed that you notice the memory increasing drastically? I.e. 600mb to 800mb

**Comment by gen2-sys-it** [comment]
> @gen2-sys-it any specific action being performed that you notice the memory increasing drastically? I.e. 600mb to 800mb

No. That's it's startup memory usage. Do you have a debug version that might help figure what is going on?

**Comment by gen2-sys-it** [comment]
@Ross-Patterson Any ideas?

**Comment by Ross-Patterson** [comment]
This is going to be tough to debug:
Can you let me know how many Portals you have open and number of files in each Portal?

Also rough idea of the file types in each Portal?

And any massive files in any portal? (5gb+)

Maybe a quick screen recording showing your setup, you can send it to: contact@rosspat.dev if you do not want to post it here.

**Comment by SonicRay100** [comment]
This is how I use Portals:

I have a folder on the root of my C: drive called Portals and each folder inside that corresponds to a Portal on my desktop. These folders are filled with nothing but shortcuts, so I don't have any large files at all.

This keeps my Portals organised. I never use it to directly link to a random folder. If I need to make a new Portal for something, I always create a new folder in my Portals folder and fill it with shortcuts.

I have 12 Portals so far on my second monitor with a total of 178 shortcuts spread across them. Highest amount is 45 within a single Portal.

Currently was sitting at 81.5 MB RAM used, but I haven't interacted with them much today. I Relauched Portals and it was using 156.7 MB RAM at start so perhaps (in my case anyway) the memory usage drops after a while?

**Comment by Ross-Patterson** [comment]
@SonicRay100 You are using how I would expect most people to use it. I use it the same way.

The numbers you gave (150mb peak) seem reasonable to me tbh. Unless I build from the ground up in plain C++ the likelihood of achieving <100mb performance consistently is slim.

I'm more interested in the 600-800mb cases and what is the setup at that point.

**Comment by SonicRay100** [comment]
Yeah I don't mind the memory usage that Portals is using but you asked for use cases so I provided mine. Granted I don't interact with the portals constantly so the memory usage never really gets that high. I also shut down my computer every night so that might be a factor for other people.

But I can get the memory usage to be high if I do constantly interact with the portals, which is what happened when I was setting them up and the memory usage peaked at 600 MB+.

Now I don't know if other people are using their portals extensively and that's why their memory usage is high all the time or if there are some other variables, but this is just my experience. So no, the memory leak is not really a problem for me as I've been now using Portals for months after initially opening this issue and the memory usage is pretty low for the most part. But the more use cases you know about the better right? :)
